src/chainparams.cpp

```diff
@@ -105,6 +105,30 @@ static CBlock FindDevNetGenesisBlock(const CBlock &prevBlock, const CAmount& rew
     assert(false);
 }
 
+bool CChainParams::UpdateMNActivationParam(int nBit, int height, int64_t timePast, bool fJustCheck) const
+{
+    for (int index = 0; index < Consensus::MAX_VERSION_BITS_DEPLOYMENTS; ++index) {
+        if (consensus.vDeployments[index].bit == nBit) {
+            auto& deployment = consensus.vDeployments[index];
+            if (timePast > deployment.nTimeout) {
+                LogPrintf("%s: activation by bit=%d for deployment %s timed out at height=%d\n", __func__, nBit, VersionBitsDeploymentInfo[Consensus::DeploymentPos(index)].name, height);
+                continue;
+            }
+            if (deployment.nMNActivationHeight < 0) {
+                LogPrintf("%s: trying to set MnEHF height=%d for non-masternode activation fork bit=%d\n", __func__, height, nBit);
+                return false;
+            }
+            if (!fJustCheck) {
+                LogPrintf("%s: set MnEHF height=%d for bit=%d successfuly while fJustCheck=%d\n", __func__, height, nBit, fJustCheck);
+                deployment.nMNActivationHeight = height;
+            }
+            return true;
+        }
+    }
+    LogPrintf("%s: not found MnEHF fork bit=%d\n", __func__, nBit);
```

Comment:
nit: Let's maybe make it a bit more loud
```suggestion
    LogPrintf("%s: WARNING: unknown MnEHF fork bit=%d\n", __func__, nBit);
```